ehr/api-src/org/labkey/api/ehr/demographics/SourceDemographicsProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2013-2016 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.labkey.api.ehr.demographics;
+
+import org.labkey.api.module.Module;
+import org.labkey.api.query.FieldKey;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * User: bimber
+ * Date: 7/9/13
+ * Time: 10:12 PM
+ */
+public class SourceDemographicsProvider extends AbstractListDemographicsProvider
```

Comment:
Similar question - why copy this to the EHR module if nobody references it? I understand we want to more the core EHR module more functional, but I'd rather prioritize the items where we have a concrete need instead of pulling misc classes in the hopes they might be useful for a hypothetical future customer. Or did I miss usage in JHU?

Reply:
This is being used in JHU as a registered demographics provider.